haystack/preview/components/reader/extractive.py

```diff
@@ -0,0 +1,321 @@
+from pathlib import Path
+from typing import List, Optional, Tuple, Union
+import math
+from haystack.preview import component, Document, ExtractedAnswer
+from haystack.preview.lazy_imports import LazyImport
+
+with LazyImport(message="Run 'pip install farm-haystack[inference]'") as torch_and_transformers_import:
+    from transformers import AutoModelForQuestionAnswering, AutoTokenizer
+    from tokenizers import Encoding
+    import torch
+
+
+@component
+class ExtractiveReader:
+    """
+    A component for performing extractive QA
+    """
+
+    def __init__(
+        self,
+        model: Union[Path, str] = "deepset/roberta-base-squad2-distilled",
+        device: Optional[str] = None,
+        top_k: Optional[int] = None,
+        top_p: Optional[float] = None,
+        max_seq_length: int = 384,
+        stride: int = 128,
+        max_batch_size: Optional[int] = None,
+        answers_per_seq: Optional[int] = None,
+    ) -> None:
+        """
+        Creates an ExtractiveReader
+        :param model: A HuggingFace transformers question answering model.
+            Can either be a path to a folder containing the model files or an identifier for the HF hub
+            Default: `'deepset/roberta-base-squad2-distilled'`
+        :param device: Pytorch device string. Uses GPU by default if available
+        :param top_k: Number of answers to return per query
+            Default: 10
```

Comment:
I think it would be nice to say something like `If not set the default will be 10` if the current formatting is not a hard requirement. Otherwise, it is a little confusing that the function signature has `None` as default.

Reply:
You are right. It used to have 10 as default, but it got a bit more complicated when `top_p` was introduced and I forgot to update the doc string.